### DISCUSSION TITLE
Support for JWT Claim-Based Rules in Mockoon

### DISCUSSION THREAD

**Discussion by namlegief** [mockoon/mockoon#1540] [Help]
I would like to request the ability to create rules in Mockoon based on JWT claims.
 Specifically, I need the ability to check the groups or Roles claim within a JWT token and return a 401 Unauthorized response if certain groups are present or missing in the JWT payload.

Use Case: The feature would allow more dynamic response handling based on JWT content, which is essential for simulating authentication and authorization flows in mock APIs. For example:

If a JWT token contains a groups claim, I want to return a 401 Unauthorized status if specific groups are either missing or not authorized.
Proposed Solution: Enable the use of JWT claims in the rules section of Mockoon. This would include:

The ability to extract and evaluate JWT claims directly in the rule conditions.
Allow logical operations to verify if a claim contains specific values (e.g., checking if the groups claim includes a specific group).
Benefits:

This feature would greatly enhance Mockoon’s ability to simulate complex authentication and authorization scenarios.
It would make it easier for developers to mock secured APIs that rely on JWT-based access controls.
Current Workaround: Currently, there is no straightforward way to achieve this behavior in Mockoon. While JWT payloads can be extracted in the response, there is no way to create rules or conditions based on their content.

I would appreciate your consideration of this feature for a future release.

**Comment by 255kb** [comment]
Hi, I suspect that more and more specific rules like that will get requested in the future. I don't think it's a good thing to create many type of specific rules, especially when we already have "Header" which is more or less targeting the same source. However, I agree that it's nearly impossible to do what you suggest with what we currently have. 
So, I've created a new type of rule that lets you enter arbitrary templating and compare it to a value. Like this, it will enable many more use cases, with the tradeoff of having to write a bit of templating. 

The issue is [here](https://github.com/mockoon/mockoon/issues/1545) and it will look like this: 

![image](https://github.com/user-attachments/assets/022a5541-65ad-4c27-9ebe-13bbbb401980)

After that we can see if the system needs more improvements.

**Reply by namlegief** [reply]
@255kb 
Hi! Thank you for the quick reply. I appreciate the fast turnaround. 
How can I get access to this version to test it and provide feedback?

Best regards,
Evgeny

**Reply by 255kb** [reply]
There is currently no alpha/beta channel. You would have to build from the sources (see contributing.md file). I can trigger a CI to build the binaries, except if you are using Windows as we are in the process of changing our code signing process.

**Reply by namlegief** [reply]
Hi @255kb I use a Mac, Mockoon 8.4.0 installed with homebrew.

**Reply by 255kb** [reply]
Pipeline is [running](https://github.com/mockoon/mockoon/actions/runs/11256667732). You should be able to access the dmg files when it finishes (tell me if it's not the case). However, you won't be able to get it through Homebrew, and I'm not sure if installing from dmg messes up with Homebrew future updates (never really used Homebrew, sorry!).

**Reply by namlegief** [reply]
Wow! Thanks a lot, worth case scenario, 
I will delete the homebrew installation and install it from DNG.
